src/plugins/intel_cpu/src/emitters/tpp/common/kernel_executors/brgemm.cpp

```diff
@@ -0,0 +1,158 @@
+    if (config.is_empty())
+        return compiled_kernel;
+
+    libxsmm_gemm_shape m_shape = libxsmm_create_gemm_shape(config.get_N(),
+                                                           config.get_M(),
+                                                           config.get_K(),
+                                                           config.get_LDB(),
+                                                           config.get_LDA(),
+                                                           config.get_LDC(),
+                                                           config.get_type_in0(),
+                                                           config.get_type_in1(),
+                                                           config.get_type_out0(),
+                                                           config.get_type_exec());
+    compiled_kernel->brgemm_kernel =
+        std::make_shared<libxsmm_gemmfunction>(reinterpret_cast<const libxsmm_gemmfunction>(COMPILE_TPP_KERNEL(
+            libxsmm_dispatch_gemm(m_shape, config.get_compile_flags(), config.get_prefetching_flags()))));
+
+    return compiled_kernel;
+}
+
+void BrgemmKernelExecutor::update_config(const ov::snippets::lowered::ExpressionPtr& expr,
+                                         const ov::snippets::lowered::LinearIRCPtr& linear_ir,
+                                         BrgemmKernelConfig& config) const {
+    int64_t M, N, K, beta;
+    std::tie(M, N, K, beta) = BrgemmKernelExecutorHelper::get_runtime_brgemm_params(expr, linear_ir);
+    const auto& tpp_mod = std::dynamic_pointer_cast<tpp::modifier::TensorProcessingPrimitive>(expr->get_node());
+    auto replace_full_dim = [](size_t dim, size_t replace_dim) {
+        if (ov::snippets::utils::is_full_dim_value(dim))
+            return replace_dim;
+        return dim;
+    };
+
+    const auto num_ins = expr->get_node()->get_input_size();
+    const auto num_outs = expr->get_node()->get_output_size();
+
+    std::vector<size_t> io_strides(num_ins + num_outs);
+
+    for (size_t i = 0; i < num_ins; i++) {
+        io_strides[i] =
+            replace_full_dim(tpp_mod->get_input_stride(i), expr->get_input_port_descriptor(i)->get_shape().back());
+    }
+
+    for (size_t i = 0; i < num_outs; i++) {
+        const auto i_off = i + num_ins;
+        io_strides[i_off] =
+            replace_full_dim(tpp_mod->get_output_stride(i), expr->get_output_port_descriptor(i)->get_shape().back());
+    }
+
+    config.update(M, N, K, io_strides[0], io_strides[1], io_strides[2], beta);
+    // update compile flag, which is depend on beta. should be part of hash.
+    config.set_compile_flags_with_zero_beta(config.get_beta() == 0);
+}
+
+void BrgemmKernelExecutor::execute(const BrgemmKernelExecutor* executor, void* in0, void* in1, void* out0) {
+    OV_CPU_JIT_EMITTER_ASSERT(executor, "has nullptr executor");
+    libxsmm_gemm_param gemm_p;
+    gemm_p.a.primary = in1;
+    gemm_p.b.primary = in0;
```

Comment:
`in0` and `in1` look like mixed up: I'd say that A input should be `in0`, not `in1`. However, in x64 impl, there is the same situation... Do you have any idea why it is done in this way?
@IvanNovoselov or maybe you have a secret TPP knowledge why we form runtime args in such way? :)

Reply:
I think the data is `row major` in ov, MatMul in libxsmm assume data is `column major`. Exchange `in0 `and `in1 `could avoid data repack. The M and N are also exchanged in `libxsmm_create_gemm_shape()`. @IvanNovoselov could you confirm it or correct me if I misunderstand it. Thank you!

Reply:
Aha, got it, thanks for the explanation! Maybe we could leave an explanatory comment then to avoid potential questions in the future?

Reply:
comment added!